pmlc/ast/builder.cc

```diff
@@ -750,6 +752,44 @@ struct ProgramBuilder {
         .result();
   }
 
+  Value makeLoopOp(ExprNodeIntrinsic *node, ArrayRef<Value> operands) {
+    if (operands.size() < 1) {
+      throw std::runtime_error(
+          "'index' primitive expects at least one operand");
+    }
+    Value loopCycle = operands.front();
+    std::vector<ExprNodePtr> loopOperation(node->operands.begin()+1, node->operands.end());
+    AstTraversal traversal(loopOperation);
+
+    // replace layer op to loop op.
+    auto layerOp = builder.create<layer::BoxOp>(
```

Comment:
So we need to create scf.for here.

Reply:
yes, if we follow yao's PR , we need create scf.for here.